machinery-helper/machinery_helper.go

```diff
@@ -0,0 +1,231 @@
+        case "l":
+          files[fileName] = linkTarget
+      }
+    }
+  }
+
+  addImplicitlyManagedDirs(dirs, files)
+
+  return files, dirs
+}
+
+func assembleJson(unmanagedFilesMap interface{}) string {
+  jsonMap := map[string]interface{}{ "extracted": false, "files": unmanagedFilesMap }
+  json, _ := json.MarshalIndent(jsonMap, " ", "  ")
+  return string(json)
+}
+
+var readDir = func(dir string) ([]os.FileInfo, error) {
+  return ioutil.ReadDir(dir)
+}
+
+func hasManagedDirs(dir string, rpmDirs map[string]bool) bool {
+  for rpmDir := range rpmDirs {
+    if strings.HasPrefix(rpmDir, dir + "/") {
+       return true
+    }
+  }
+  return false
+}
+
+func findUnmanagedFiles(dir string, rpmFiles map[string]string, rpmDirs map[string]bool, unmanagedFiles map[string]string) {
+  files, _ := readDir(dir)
+  for _, f := range files {
+    fileName := dir + f.Name()
+    if !utf8.ValidString(fileName) {
+      fmt.Fprintln(os.Stderr, fileName, "contains invalid UTF-8 characters. Skipping.")
+    } else {
+      if _, ok := IgnoreList[fileName]; !ok {
+        if f.IsDir() {
+          if _, ok := rpmDirs[fileName]; ok {
+            findUnmanagedFiles(fileName + "/", rpmFiles, rpmDirs, unmanagedFiles)
+          } else {
+            if !hasManagedDirs(fileName, rpmDirs) {
+              unmanagedFiles[fileName + "/"] = "dir"
+            }
+          }
+        } else {
+          if _, ok := rpmFiles[fileName]; !ok {
+            if f.Mode() &
+              (os.ModeSocket | os.ModeNamedPipe | os.ModeDevice | os.ModeCharDevice) != 0 {
+              // Ignore sockets, named pipes and devices
+            } else if f.Mode() & os.ModeSymlink == os.ModeSymlink {
+              unmanagedFiles[fileName] = "link"
+            } else {
+              unmanagedFiles[fileName] = "file"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+func printVersion() {
+  fmt.Println("Version:", VERSION)
+  os.Exit(0)
+}
+
+func main() {
+  // check for tar extraction
+  if len(os.Args) >= 2 {
+    switch os.Args[1] {
+    case "tar":
+      Tar(os.Args[2:])
+      os.Exit(0)
+    }
+  }
+
+  // parse CLI arguments
+  var versionFlag = flag.Bool("version", false, "shows the version number")
+  flag.Parse()
+
+  // show version
+  if (*versionFlag == true) {
+    printVersion()
+  }
+
+  // fetch unmanaged files
+  unmanagedFiles := make(map[string]string)
+  thisBinary, _ := filepath.Abs(os.Args[0])
+
+  ignoreList := map[string]bool{
+    thisBinary: true,
+  }
+  for _, mount := range RemoteMounts() {
+    ignoreList[mount] = true
+  }
+  for _, mount := range SpecialMounts() {
+    ignoreList[mount] = true
+  }
+  IgnoreList = ignoreList
+
+  for _, mount := range RemoteMounts() {
+    unmanagedFiles[mount + "/"] = "remote_dir"
+  }
+
+  rpmFiles, rpmDirs := getManagedFiles()
+  findUnmanagedFiles("/", rpmFiles, rpmDirs, unmanagedFiles)
+
+  files := make([]string, len(unmanagedFiles))
+  i := 0
+  for k, _ := range unmanagedFiles {
```

Comment:
should omit 2nd value from range; this loop is equivalent to `for k := range ...`